### PR TITLE
Pkg 5252 shapely 2.0.5

### PR DESCRIPTION
**Pull request by Jrice1317**
shapely 2.0.5 ❄️ 

**Destination channel:** {defaults}

### Links

- [PKG-5252](https://anaconda.atlassian.net/browse/PKG-5252) 
- [Upstream repository](https://github.com/shapely/shapely/tree/2.0.5)
- [Upstream changelog/diff](https://github.com/shapely/shapely/blob/2.0.5/CHANGES.txt)

[PKG-5252]: https://anaconda.atlassian.net/browse/PKG-5252?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -46,7 +46,10 @@ test:
     - shapely.vectorized
   commands:
     - pip check
-    - pytest --pyargs shapely
+    # osx fails if LC_ALL is not Italian or Portuguese
+    # see https://github.com/shapely/shapely/blob/2.0.5/shapely/tests/legacy/test_locale.py#L14
+    - export LC_ALL=it_IT.UTF-8  # [osx]
+    - pytest shapely/tests -v
```
Comment (cbouss) [inline recipe/meta.yaml]: You should be able to remove the source_files section and run `pytest --pyargs shapely`, as the tests are included in the package itself. This avoids having a copy of the tests in the info folder.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -27,15 +27,15 @@ requirements:
     - numpy {{ numpy }}
```
Comment (cbouss) [inline recipe/meta.yaml:27]: The pinning on cython could be removed.

**Review by cbouss — APPROVED** [review]
LGTM, some suggestions but nothing major.

File: recipe/meta.yaml (hunk 1 above)
Comment (psteyer) [reply recipe/meta.yaml]: How are you able to tell if the tests are included in the package itself?
I end up using `source_files` a lot, and am now wondering if there are times I actually don't need to.

Comment (cbouss) [reply recipe/meta.yaml]: Here I see them nested in the shapely package: https://github.com/shapely/shapely/tree/2.0.5/shapely/tests

**Review by danpetry — APPROVED** [review]

**Pull request merged**